src/WS.cpp

```diff
@@ -1773,245 +1983,438 @@ signed char WS::root_callback(struct lejp_ctx *ctx, char reason)
     return 0;
 }
 
+void restart_threads_by_signal(int &flag)
+{
+
+    // inform main to restart threads
+    std::unique_lock lck(mutex_main);
+    if ((flag & PNT_FLAG_RESTART_RTSP) || (flag & PNT_FLAG_RESTART_VIDEO) || (flag & PNT_FLAG_RESTART_AUDIO))
+    {
+        if (flag & PNT_FLAG_RESTART_RTSP)
+        {
+            global_restart_rtsp = true;
+            flag &= ~PNT_FLAG_RESTART_RTSP;
+        }
+        if (flag & PNT_FLAG_RESTART_VIDEO)
+        {
+            global_restart_video = true;
+            flag &= ~PNT_FLAG_RESTART_VIDEO;
+        }
+        if (flag & PNT_FLAG_RESTART_AUDIO)
+        {
+            global_restart_audio = true;
+            flag &= ~PNT_FLAG_RESTART_AUDIO;
+        }
+        global_cv_worker_restart.notify_one();
+    }
+}
+
+std::string generateSessionID()
```

Comment:
So far we've usually been using `Id` for other identifiers.

Reply:
first use of 'id' in this project, but can be renamed to session_id. Whats your opinion on this ? @gtxaspec , @themactep

Reply:
Yeah, I meant in comparison with `devId` or `clientSessionId` that already exist in the code from a code style naming perspective.

Reply:
In relation to the `clientSessionId`, maybe we should make it more clear this is a `wsSessionId` ?